flang/runtime/transformational.cpp

```diff
@@ -521,9 +521,10 @@ void RTNAME(Unpack)(Descriptor &result, const Descriptor &vector,
     }
     result.IncrementSubscripts(resultAt);
     mask.IncrementSubscripts(maskAt);
-    field.IncrementSubscripts(fieldAt);
+    if (field.rank() != 0) { // scalar "field" argument
+      field.IncrementSubscripts(fieldAt);
+    }
```

Comment:
Did you already add this change to llvm-project? 

@schweitzpgi is it better to put this change in a separate PR if it is already added to llvm-project? 

Or can this change wait until fir-dev is upstreamed to llvm-project?

Reply:
Normally, we just cherry pick over any code from llvm-project, keeping the original change as intact as is possible. For example, we wouldn't squash the cherry-picked changes with a bunch of other changes. That helps both git and humans figure out what is happening. :)

Caveat: it may be that only part of an llvm-project commit can be cherry-picked. That should be fine as well. What we are really trying to avoid are make-work sorts of conflicts.

Reply:
But in this case, I don't see this change in the "main" branch on llvm-project.

Reply:
Should it be? Or this is completely new work that hasn't been merged into llvm-project yet?

Either way, we can merge it into fir-dev. We just want to keep these changes "in synch" somehow, right?

Reply:
The main reason for this change is to improve our results on the tests that Steve Scalpone is running, which I assume are the result of building the compiler from fir-dev.

I was not personally planning to move these changes into llvm-project.